src/software/new_geom/line.h

```diff
@@ -0,0 +1,99 @@
+#pragma once
+
+#include "software/new_geom/point.h"
+#include "software/new_geom/polynomial.h"
+
+/**
+ * A 2D line.
+ */
+class Line final
+{
+   public:
+    /**
+     * Creates a degenerate Line with zero Points and zero Polynomials
+     */
+    Line();
+
+    /**
+     * Creates a Line from two Points
+     *
+     * @param first the first Point
+     * @param second the second Point
+     */
+    explicit Line(const Point& first, const Point& second);
+
+    /**
+     * Returns the first Point
+     *
+     * @return the first Point
+     */
+    Point getFirst() const;
+
+    /**
+     * Returns the second Point
+     *
+     * @return the second Point
+     */
+    Point getSecond() const;
```

Comment:
Line is infinitely long so getting the first and second point doesn't make a lot of sense

Reply:
Agreed, we should just be able to query a point on the line. But this gets tricky because it gets very difficult to get some context for where `t=0` is in 2D space.

Reply:
New implementation removes these accessors